From 45c259b1922746baecff287015e2159bb1e32c10 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Thu, 10 Jul 2014 00:58:29 -0400
Subject: [acpi] dock: avoid initializing acpi_dock_notifier_list multiple times

Message-id: <20140710005828.22146.51178.stgit@gir.stowe>
Patchwork-id: 85541
O-Subject: [RHEL7.1 PATCH 002/105] ACPI / dock: avoid initializing acpi_dock_notifier_list multiple times
Bugzilla: 1114228
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1114228
Upstream Status: f22ff5523af8b365167cb79189f8b91470d57c8c

commit f22ff5523af8b365167cb79189f8b91470d57c8c
Author: Jiang Liu <jiang.liu@huawei.com>
Date:   Sat Jun 29 00:24:34 2013 +0800

    ACPI / dock: avoid initializing acpi_dock_notifier_list multiple times

    Function dock_add() will be called multiple times if there are
    multiple dock stations, which causes acpi_dock_notifier_list to be
    initialized multiple times.

    To avoid that, move the initialization of acpi_dock_notifier_list
    from dock_add() to acpi_dock_init().

    [rjw: Changelog]
    Signed-off-by: Jiang Liu <jiang.liu@huawei.com>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/acpi/dock.c b/drivers/acpi/dock.c
index 8265607..c36de86 100644
--- a/drivers/acpi/dock.c
+++ b/drivers/acpi/dock.c
@@ -1007,7 +1007,6 @@ static int __init dock_add(acpi_handle handle)
  mutex_init(&dock_station->hp_lock);
  spin_lock_init(&dock_station->dd_lock);
  INIT_LIST_HEAD(&dock_station->sibling);
- ATOMIC_INIT_NOTIFIER_HEAD(&dock_notifier_list);
  INIT_LIST_HEAD(&dock_station->dependent_devices);
 
  /* we want the dock device to send uevents */
@@ -1078,6 +1077,7 @@ void __init acpi_dock_init(void)
   return;
  }
 
+ ATOMIC_INIT_NOTIFIER_HEAD(&dock_notifier_list);
  register_acpi_bus_notifier(&dock_acpi_notifier);
  pr_info(PREFIX "%s: %d docks/bays found\n",
   ACPI_DOCK_DRIVER_DESCRIPTION, dock_station_count);
-- 
1.7.1